src/components/FetchApi.js

```diff
@@ -1,52 +1,131 @@
+        fetch('https://jsonplaceholder.typicode.com/posts')
+            .then((resp) => resp.json())  //the fetched data is parsed into json format.
+            .then((postData) => this.setPost(postData)); // Calling setUsers and passing the data.
+    }
+
+    addPost(e) {
+        e.preventDefault()
+        let title = document.getElementById('title').value;
+        let body = document.getElementById('body').value;
+        fetch('https://jsonplaceholder.typicode.com/posts', {
+            method: 'POST',
+            headers: {
+                'Accept': 'application/json , text/plain , */*',
+                'Content-type': 'application/json'
+            },
+            body: JSON.stringify({ title: title, body: body })
+    
+        }).then((res) => res.json())
+        .then((data) => this.setAddPost(data))
+    }
+    
+
+  
+    setUsers = (data) => {
         //created a new function to set the state of users
         this.setState({
-            users:data
+            users: data
         })
     }
 
-    
-  renderUser=()=> {
-    return  this.state.users.map((user) => {
-            return <div class="user">
-        <h3>User Id: ${user.id}</h3>
-         <span className="userInfo">Name: ${user.name}</span>
-         <span className="userInfo userName">User Name: ${user.username}</span>
-         <span className="userInfo">Address: ${user.address.street},${user.address.city},${user.address.suite}</span>
-         <span className="userInfo">Email: ${user.email}</span>
-         <span className="userInfo">Contact: ${user.phone}</span>
-         <span className="userInfo">Website-link: ${user.website}</span>
-        </div> })
-    //we have removed join from here because in react everything is javascript and we are using JSX.
-      }
+    setPost = (postData) => {
+        this.setState({
+            posts: postData
+        })
+    }
 
+    setAddPost=(addPostTitle,addPostBody)=>{
+        this.setState({
+            title:addPostTitle,
+            body:addPostBody
+        })
+    }
+
+  //Create a functional component for rendering users 
+    renderUser = () => {
+        return this.state.users.map((user) => {
+            return <div className="user">
+                <h3>User Id: {user.id}</h3>
+                <span className="userInfo">Name: {user.name}</span>
+                <span className="userInfo userName">User Name: {user.username}</span>
+                <span className="userInfo">Address: {user.address.street},{user.address.city},{user.address.suite}</span>
+                <span className="userInfo">Email: {user.email}</span>
+                <span className="userInfo">Contact: {user.phone}</span>
+                <span className="userInfo">Website-link: {user.website}</span>
+            </div>
+        })
+        //we have removed join from here because in react everything is javascript and we are using JSX.
+    }
+
+    renderPost = () => {
+        return this.state.posts.map((post) => {
```

Comment:
create a separate component for rendering posts, pass it as children.
read up on how to use children props and make the changes
check the react docs for children props or check your Javascript course